### PR TITLE
bpf: do not drop unknown protos in nat handling

### PR DESCRIPTION
**Pull request by borkmann**
See commit msg @brb

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/10526)

### CODE REVIEW

**Comment by borkmann** [comment]
test-me-please

**Comment by borkmann** [comment]
test-me-please

**Comment by borkmann** [comment]
provision failure, retrying

**Comment by borkmann** [comment]
test-me-please

File: bpf/lib/nat.h [hunk 1]
```diff
@@ -465,9 +465,7 @@ static __always_inline __maybe_unused int snat_v4_create_dsr(struct __ctx_buff *
 		tuple.sport = l4hdr.dport;
 		break;
 	default:
-		// NodePort svc can be reached only via TCP or UDP, so
-		// drop the rest
-		return DROP_NAT_UNSUPP_PROTO;
+		return NAT_PUNT_TO_STACK;
```
Comment (brb) [inline bpf/lib/nat.h]: `NAT_PUNT_TO_STACK` is `< 0`, so the caller of `snat_v4_create_dsr()` will drop such packet: https://github.com/cilium/cilium/blob/master/bpf/lib/nodeport.h#L819-L820

**Review by brb — CHANGES_REQUESTED** [review]

**Comment by brb** [comment]
I'm wondering what would be the easiest way to test it in CI - maybe sending ICMP timestamp request with `hping3 127 $NODE_IP --icmp --icmp-ts` from the third node?

File: bpf/lib/nat.h (hunk 1 above)
Comment (borkmann) [reply bpf/lib/nat.h]: Hm, but that is okay no? We know from handle_dsr_v4() that there was an IP option for DSR present, so anything else would be bad not to drop. I'll leave it at `DROP_NAT_UNSUPP_PROTO` here.

**Comment by borkmann** [comment]
test-me-please

**Review by brb — APPROVED** [review]

**Comment by brb** [comment]
Created the issue for CI tests: https://github.com/cilium/cilium/issues/10541.

**Pull request merged**